web-app/HeimScripts/heatmap/limmaUtils.R

```diff
@@ -33,3 +29,25 @@ getDesign <- function(measurements) {
 getSubset1Length <- function(measurements) {
   sum(grepl(pattern = SUBSET1REGEX, x = colnames(measurements))) # returns number of column names satisfying regexp.
 }
+
+# to check if a subset contains at least one non missing value
+isSubsetHasNonNA <- function (subset, measurements) {
+   # select the  measurements of a subset by matching the subset label with column names
+   # each measurements column has subset information as suffix
+   # eg: X1000314002_n0_s2, X1000314002_n0_s1
+   subsetMeasurement <- measurements[,grep(paste(c(subset, '$'), collapse=''), names(measurements))]
+   # check if there's non missing values
+   sum(!is.na(subsetMeasurement)) > 0
+}
+
+# to check if row  contains at least one row that contains 3 non missing values
+rowContainsAtLeastThreeData <- function (row) {sum(!is.na(row)) > 2}
+
+# measurements should contain at least :
+# - one valid row
+# - both subsets contains at least one non NA
+isValidLimmaMeasurements <- function (measurements) {
+   sum(apply(measurements, 1, rowContainsAtLeastThreeData)) > 0 & # check one valid row
```

Comment:
I thought that it suffice if there is at least one valid row - not every row needs to have at least1 value.